addons/web_editor/static/src/js/wysiwyg/widgets/link_popover_widget.js

```diff
@@ -59,15 +67,46 @@ const LinkPopoverWidget = Widget.extend({
             // 3. Remain open when the popover content is clicked..
             // 4. ..except if it the click was on a button of the popover content
             // 5. Close when the user click somewhere on the page (not being the link or the popover content)
-            trigger: 'focus',
+            trigger: 'manual',
```

Comment:
Not sure if this is still a WIP or not (since not in draft) but please triple check in website, as this broke everything.
There is a long history about this trigger which was changed a loooooot of time to have a stable behavior (reason why I added the comment above)
Let me know if you need me to test again once PR is done

Reply:
The PR is done.
The problem with `'focus'` is that it does not trigger in the mass_mailing.
I have tested it to respect the spec of the comment above in note, website and mass mailing.

Reply:
It is still very buggy in the website.
See https://watch.screencastify.com/v/FPuJElh1ysixMfBxURb7 and https://watch.screencastify.com/v/OK0thhKTa2BcyLM6AVVp
In one case the tooltip is not opening on button click (well it is but closed directly so looks like it is not)
In the other case, on latter click on link, the popover it closed (shouldnt be) and can't be reopened

Also https://watch.screencastify.com/v/egpEWWnMgnoIDXZO6FLw
Create a link > click on link > popover is not shown

Reply:
Actually, the glitch that you see comes from another bug that I fixed in a PR that depends on this one.

In this PR, I add the behavior where the tooltip disappear if the selection is not collapsed. The reason being that otherwise you can have the floating toolbar in a field html being on top of the tooltip (which is strange visually and functionally).

The other bug is changing the selection although it shouldn't. As the selection change, the tooltip disappear.

This is fixed in the following PR (that depends on this one): https://github.com/odoo/odoo/pull/78301
I've made a video from the branch of that PR: https://watch.screencastify.com/v/VjJJVcsV7qEf5qopM7L5

To check the behavior, it is better that you test it on the other branch (that include the commit of this branch).

Reply:
I tested on the other branch you mentioned, it seems to be a bit better but still a few issues:

1. Create a link > Click on the link directly (without focusing out first) > Click again on the link -> Traceback
![image](https://user-images.githubusercontent.com/30048408/140511263-09a73d04-578b-40d9-ac9f-d1dfcc325bca.png)
Somehow when I try to record this one, it crashed after a lot of click, not after 2 :shrug: but without recording, always on second click
2. When clicking on a button, the link popover is not shown in most cases, it seems to depend what you were focusing before clicking on it https://watch.screencastify.com/v/JD8tVZABWr4K9U5LmH0j
3. Click once on a link, the popover is show, then click twice on the link, popover is hidden and do not remain shown https://watch.screencastify.com/v/wupXvSQ4iGeMlthCfKyG

Reply:
1. This is a bug that does not seems directly related to this fix therefore I've made another task for it: https://www.odoo.com/web#id=2684815&action=333&active_id=1695&model=project.task&view_type=form&cids=1&menu_id=4720
2. We hide the popover when the selection is not collapsed. whenever clicking on a link with the class `o_default_snippet_text`, the link gets selected, therefore the popover disappear. The bug however is that the class `o_default_snippet_text` comes back after the first click and even after changing the text. I'm looking into it.
3. Again it is because the selection is not collapsed.

We hide the tooltip on uncollapsed selection in order to avoid having the floating toolbar in top of the tooltip. I can make a special case to allow uncollapsed selection whenever there is no floating bar. @rdeodoo what do you think ?

Reply:
I added the case to only remove the tooltip upon uncollapsed selection only when there is a floating toolbar (ie not in website and mass_mailing).

Reply:
Seems way better now thank you